### PR TITLE
`mod internal`: Add comments to internal structures

### PR DESCRIPTION
**Pull request by rinon**
Copies all comments from internal.h to our Rust version. Preparing to make core internal structures safe Rust types.

### CODE REVIEW

File: src/internal.rs [hunk 1]
```diff
@@ -322,57 +350,64 @@ impl Rav1dFrameContext_bd_fn {
...
+    // in-loop filter per-frame state keeping
     pub start_of_tile_row: *mut u8,
     pub start_of_tile_row_sz: c_int,
     pub need_cdef_lpf_copy: c_int,
     pub p: [*mut DynPixel; 3],
     pub sr_p: [*mut DynPixel; 3],
     pub mask_ptr: *mut Av1Filter,
     pub prev_mask_ptr: *mut Av1Filter,
-    pub restore_planes: c_int,
+    pub restore_planes: c_int, // enum LrRestorePlanes
```
Comment (kkysen) [inline src/internal.rs:410]:
```suggestion
    pub restore_planes: LrRestorePlanes,
```

We should be able to just use the `enum` directly.  C forces `enum`s to be `int`-sized, but Rust doesn't (though in this case, it's still `c_int`, but future `dav1d` versions could change that, as they sometimes make the type `uint8_t` to save space).

File: src/internal.rs [hunk 2]
```diff
@@ -160,6 +160,10 @@ pub(crate) struct TaskThreadData {
     pub cond: pthread_cond_t,
     pub first: atomic_uint,
     pub cur: c_uint,
+    // This is used for delayed reset of the task cur pointer when
+    // such operation is needed but the thread doesn't enter a critical
+    // section (typically when executing the next sbrow task locklessly).
+    // See src/thread_task.c:reset_task_cur().
```
Comment (kkysen) [inline src/internal.rs:166]: Do we want this to be an intra-doc link to the Rust `fn`?

**Review by kkysen — COMMENTED** [review]
Generally LGMT; it's just comments.  One question, though: do we want the comments on fields, types, etc. to be doc comments?

File: src/internal.rs (hunk 1 above)
Comment (rinon) [reply src/internal.rs:410]: Sure, but let's do that in a separate change later.

**Comment by rinon** [comment]
I don't think we should spend the time right now to make them proper doc comments while these structures are still changing a lot. At some point when the structures are safe, sure.

File: src/internal.rs [hunk 3]
```diff
@@ -160,6 +160,10 @@ pub(crate) struct TaskThreadData {
     pub cond: pthread_cond_t,
     pub first: atomic_uint,
     pub cur: c_uint,
+    /// This is used for delayed reset of the task cur pointer when
+    /// such operation is needed but the thread doesn't enter a critical
+    /// section (typically when executing the next sbrow task locklessly).
+    /// See [crate::src::thread_task::reset_task_cur].
```
Comment (kkysen) [inline src/internal.rs:166]: nit:

```suggestion
    /// See [`crate::src::thread_task::reset_task_cur`].
```

**Review by kkysen — APPROVED** [review]

**Pull request merged**